octopusdeploy_framework/resource_generic_oidc_account_test.go

```diff
@@ -0,0 +1,62 @@
+package octopusdeploy_framework
+
+import (
+	"fmt"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/core"
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"strings"
+	"testing"
+)
+
+func TestAccGenericOidcAccountBasic(t *testing.T) {
+	localName := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	resourceName := "octopusdeploy_generic_oidc_account." + localName
+
+	description := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	name := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+	tenantedDeploymentParticipation := core.TenantedDeploymentModeTenantedOrUntenanted
+
+	executionKeys := []string{"space"}
+	audience := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
+
+	config := testGenericOidcAccountBasic(localName, name, description, tenantedDeploymentParticipation, executionKeys, audience)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: ProtoV6ProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testAccountExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "description", description),
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttrSet(resourceName, "space_id"),
+					resource.TestCheckResourceAttr(resourceName, "tenanted_deployment_participation", string(tenantedDeploymentParticipation)),
+					resource.TestCheckResourceAttr(resourceName, "execution_subject_keys.0", executionKeys[0]),
+					resource.TestCheckResourceAttr(resourceName, "audience", audience),
+				),
+				ResourceName: resourceName,
+			},
```

Comment:
could you please include a 2nd `TestStep` so that we can also test update. It will basically be the same step, just with a slight change in the config, like a different description.